docs/CONFIG-VARS.md

```diff
@@ -191,7 +191,7 @@ Ubuntu 20.04 LTS is the operating system used on the Jump/NFS servers. Ubuntu cr
 | jump_rwx_filestore_path | File store mount point on jump server | string | "/viya-share" | This location cannot include `/mnt` as its root location. This disk is ephemeral on Ubuntu, which is the operating system being used for the jump/NFS servers. |
 | tags | Map of common tags to be placed on all Azure resources created by this script | map | { project_name = "sasviya4", environment = "dev" } | |
 | aks_identity | Use UserAssignedIdentity or Service Principal as  [AKS identity](https://docs.microsoft.com/en-us/azure/aks/concepts-identity) | string | "uai" | A value of `uai` wil create a Managed Identity based on the permissions of the authenticated user or use [`AKS_UAI_NAME`](#use-existing), if set. A value of `sp` will use values from [`CLIENT_ID`/`CLIENT_SECRET`](#azure-authentication), if set. |
-| ssh_public_key | File name of public ssh key for jump and nfs VM | string | "~/.ssh/id_rsa.pub" | Required with `create_jump_vm=true` or `storage_type=standard` |
+| ssh_public_key | File name of public ssh key for jump and nfs VM | string | null | Required with `create_jump_vm=true` or `storage_type=standard` |
```

Comment:
This can now be a breaking change as all our code looks for this and the users might not be used to added or adjusting this line. What/how are we replacing this default?

Reply:
TFLint throws error when setting the value to a non-existent file in `locals.tf`. I adjusted the logic in [locals.tf ](https://github.com/sassoftware/viya4-iac-azure/blob/IAC-1196/locals.tf#L7) similar to how this is set in [GCP](https://github.com/sassoftware/viya4-iac-gcp/blob/main/locals.tf#L23). GCP also sets the default value for `ssh_public_key` as null. This change wasn't synced across cloud. 

Following is the error I received from TFLint:
```
Error: Invalid function argument

  on locals.tf line 8, in locals:
   8:     ? file(var.ssh_public_key)

with var.ssh_public_key as "~/.ssh/id_rsa.pub".

Invalid value for "path" parameter: no file exists at "~/.ssh/id_rsa.pub"; this function works only with files that are distributed as part of the configuration source code, so if this file will be created by a resource in this configuration you must instead obtain this result from an attribute of that resource.

Error: Process completed with exit code 1.
```
We have always stated this as required when `create_jump_vm=true` or `storage_type=standard`. Also all our sample files  have this variable set. But you might be right because there was a default in place before folks might have overlooked at this requirement.
This one is tricky because we do not generate keypair and rather depend on user to provide it.
Will reach out to you to discuss more about this.

Reply:
I think over the short-term we should leave this and have the exception setup in the .tflint.hcl file to over-ride this error. The information is correct, but we'd need to come up with a more elegant solution in fixing this issue given its past usage. Or simply change this when we're going to have our next major release with other breaking changes.

Reply:
The changes for variable `ssh_public_key` were backed out. TFLint does not seem to be throwing the error previously received. Pushing up for re-review.